### DISCUSSION TITLE
getting_started/step_by_step/nodes_and_scenes

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#72] [User-contributed notes]
# getting_started/step_by_step/nodes_and_scenes

In Overview of Godot's key concepts, we saw that a Godot game is a tree of scenes and that each scene is a tree of nodes. In this lesson, we explain a bit more about them. You will also create your...

https://docs.godotengine.org/en/stable/getting_started/step_by_step/nodes_and_scenes.html

**Comment by scotthawkins** [comment]
About project directory only able to be created in home directory when "Creating your first scene"
Running on Linux Mint Cinnamon.
Installed Godot Engine from Software Manager resulting in installation of version 4.3 as Flatpak (Flathub).
Read all of ABOUT, and GETTING STARTED up to Step by step->Nodes and Scenes->Creating your first scene
Opened and read "create a new project"
Created new project and attempted to select different directory by browsing, since default is top level of home directory.
Most directories shown were disabled for navigation or selection except puzzling exceptions such as hidden directories starting with dot.
I resisted, since I don't want to clutter my home with individual project directories for a specific application, but could find no setting to allow a different directory.
Eventually I gave in and created a new project with the default name.
The Editor was shown and presented menus including Editor->Editor Settings...
Scrolling down I discovered FileSystem->Directories->Default Project Path
I changed Default Project Path to contain my preferred directory and then closed the project.
I removed the project from the list of projects in the Project Manager Project Tab.
This did not deleted the project directory at the top level of my home directory, so I deleted that manually.
After that I was able to create a new project in my desired location.
I found the settings file at ~/.var/app/org.godotengine.Godot/config/godot/editor_settings-4.3.tres
I'm not quite sure how much of this specific to Linux, etc,, but I imagine it might be common that it is required to create a throw-away project just to change this setting the first time.  
Perhaps since this is a tutorial for first time use that could be explained or perhaps this could be added to the quick settings in Project Manager.

**Reply by stinkysoap** [reply]
this is mostly just a problem of editor or engine not having access to other directories because flatpaks are packed in a secure passage

**Comment by hxlcd** [comment]
Under the Node section, it's mentioned that "As we saw before, you can build a playable character [...]"

I am following the docs in sequence and don't remember being shown that. I may have missed it or forgot.

I think turning the 'as we saw before' or just the word 'before' into a hyperlink to the specific part of the docs showing what's stated would clear any confusion for learners.

**Reply by urie2048** [reply]
They are referencing the example of a player character used in the overview that is linked in the intro of this page.

**Comment by gurselx** [comment]
please put it in english

**Reply by Calinou** [reply]
You can change the documentation language by clicking the icon in the middle in the bottom-right corner of the page:

<img width="249" height="76" alt="image" src="https://github.com/user-attachments/assets/372a27a2-0492-492e-9b53-a1aacd9795af" />

Set it to `en` to switch to English. If this doesn't work, change the page URL to replace the language code with `en` (it's before the version identifier).

Note that there's an issue with the resulting URL when using the Read the Docs language switcher currently, which is being fixed by https://github.com/godotengine/godot-docs/pull/11349.

**Comment by unknownreader** [comment]
How to seperate nodes that i've been made ?

**Reply by Calinou** [reply]
What do you mean by "separate nodes"?

If you want to save part of the node tree to a different file (so it can be reused in different nodes/scenes), right-click the node in question and choose **Save Branch as Scene**.

**Comment by tppt** [comment]
Something to note that really isn't spelled out in the docs here (and perhaps this is just a me problem), but when I first opened the 2D editor and added my text node, the top left corner of the node wasn't at 0,0 in the coordinate system, but at something like -2500, -1500.  When I ran my game the node text wasn't visible.  I figured out this is because what's shown when you run the game is the area in the blue outlined rectangle starting at 0,0.  Also, I feel it should be noted that the y-axis is reversed (the same as in an HTML viewport), where the positive direction is down, not up.

**Comment by MagnusOxlund** [comment]
> “[Nodes] receive callbacks to update every frame.”

This sentence can be a bit ambiguous. In my world, a “callback” is a function that’s passed around as an object to be executed later or elsewhere.

What the sentence intends to convey is that nodes may have associated functions (callbacks) that are called each frame (or at other specific times, such as in response to signals, which you can think of as events).
